core/services/ocr2/plugins/ocr2keeper/evm21/abi.go

```diff
@@ -21,6 +21,8 @@ const (
 	UPKEEP_FAILURE_REASON_UPKEEP_NOT_NEEDED
 	UPKEEP_FAILURE_REASON_PERFORM_DATA_EXCEEDS_LIMIT
 	UPKEEP_FAILURE_REASON_INSUFFICIENT_BALANCE
+	UPKEEP_FAILURE_REASON_MERCURY_CALLBACK_REVERTED
+	UPKEEP_FAILURE_REASON_MERCURY_ACCESS_NOT_ALLOWED
```

Comment:
I think we need to structure this better. There's one enum of failure reasons returned from contract and one enum of offchain failure reasons which are currently combined into a single entity with implicit mapping of first n rows

Reply:
right now, all revert reasons are 1 to 1 matched to enum onchain except the last one for access not allowed.
it's unfortunate that go wrapper does not have these enums.
and we actually redefined these in here and in ts tests.

Reply:
I'm concerned that this enum can get changed accidentally onchain or offchain breaking the implicit mapping. Maybe for now just clarify this with comments in the enum. i.e. any new reason added in contract should be added at one place and any new reason added offchain should be added in another place

Reply:
done